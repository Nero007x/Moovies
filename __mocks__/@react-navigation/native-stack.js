export const createNativeStackNavigator = jest.fn(() => ({
    Navigator: "StackNavigator",
    Screen: "Screen",
  }));
  