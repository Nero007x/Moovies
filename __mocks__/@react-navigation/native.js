export const useNavigation = () => ({
    navigate: jest.fn(),
  });
  
  export const NavigationContainer = ({ children }) => children;
  