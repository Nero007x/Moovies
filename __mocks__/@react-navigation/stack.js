export const createStackNavigator = jest.fn(() => ({
    Navigator: "StackNavigator",
    Screen: "Screen",
}));
