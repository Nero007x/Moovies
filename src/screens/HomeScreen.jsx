import React from 'react';
import { View, StyleSheet } from 'react-native';
import Header from '../components/Header';
import MovieList from '../components/MovieList';

const HomeScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Header />
      <MovieList navigation={navigation} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default HomeScreen;
