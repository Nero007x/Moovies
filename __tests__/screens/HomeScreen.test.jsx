import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../../src/screens/HomeScreen';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();

jest.mock('../../src/components/Header', () => 'Header');
jest.mock('../../src/components/MovieList', () => 'MovieList');

it('renders correctly', () => {
  const tree = renderer.create(
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

