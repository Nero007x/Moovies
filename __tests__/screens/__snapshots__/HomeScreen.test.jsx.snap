// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<StackNavigator>
  <Screen
    component={[Function]}
    name="Home"
  />
</StackNavigator>
`;
